ocs_ci/framework/pytest_customization/marks.py

```diff
@@ -392,6 +393,46 @@
     ),
     reason="Test runs ONLY on Fusion HCI provider and client clusters",
 )
+
+
+# when run_on_all_clients marker is used, there needs to be added cluster_index
+# parameter to the test to prevent any issues with the test parametrization
+def setup_multicluster_marker(marker_base, push_missing_configs=False):
+    """
+    Set up multicluster marker with parametrization based on client indexes.
+
+    Args:
+        marker_base: Base pytest marker to be parametrized
+        push_missing_configs: Boolean flag to push missing configs
+
+    Returns:
+        Parametrized marker or original marker if setup fails
+    """
+    try:
+        if push_missing_configs:
+            from ocs_ci.deployment.hosted_cluster import hypershift_cluster_factory
+
+            hypershift_cluster_factory(
+                duty="use_existing_hosted_clusters_push_missing_configs",
+            )
+        client_indexes = [
```

Comment:
is this function specific to client clusters or to multi clusters in general?

Reply:
setup_multicluster_marker is a helper function specific for multicluster scenarios, it does use significant time to execute, but we rely it. It makes possible during the execution to reduce number of client clusters in system from many to 0 or increase from 0 to many

Reply:
we have moved from this draft to https://github.com/red-hat-storage/ocs-ci/pull/12041, it is updated and verified